src/rgw/rgw_rest_swift.cc

```diff
@@ -646,13 +647,41 @@ int RGWCreateBucket_ObjStore_SWIFT::get_params()
   return get_swift_versioning_settings(s, swift_ver_location);
 }
 
+static inline int handle_metadata_errors(req_state* const s, const int op_ret)
+{
+  if (op_ret == -EFBIG) {
+    /* Handle the custom error message of exceeding maximum custom attribute
+     * (stored as xattr) size. */
+    const auto error_message = boost::str(
+      boost::format("Metadata value longer than %lld")
+        % int(s->cct->_conf->osd_max_attr_size));
+    set_req_state_err(s, EINVAL, error_message);
+    return -EINVAL;
+  } else if (op_ret == -E2BIG) {
+    const auto error_message = boost::str(
+      boost::format("Too many metadata items; max %lld")
+        % s->cct->_conf->get_val<int>("rgw_max_attrs_num_in_req"));
```

Comment:
The <int> here should almost certainly be <size_t> .  Better yet would be to use "cct->_conf->rgw_max_attrs_num_in_req".  The boost runtime code for get_val<int> generates a lot of runtime code - and for this bug a very hairy stack backtrace.

Reply:
Yeah, `boost::format` & co. isn't cheap. I hoped that having it outside of main execution paths isn't an issue.

I'm not sure whether we could use `cct->_conf->rgw_max_attrs_num_in_req` due to the recent changes in the configuration component. I guess it's allowed only for legacy options while `rgw_max_attrs_num_in_req` is a new one. Anyway, I will check this out.

Reply:
Just pushed the updated branch. It looks I introduced the `int` bug  with the rebase after the new configuration subsystem. Big thanks for pointing it out, Marcus! :-)